Tests/TurboNavigatorTests.swift

```diff
@@ -271,6 +284,14 @@ final class TurboNavigatorTests: XCTestCase {
 
 private class EmptyDelegate: TurboNavigationDelegate {
     func session(_ session: Session, didFailRequestForVisitable visitable: Visitable, error: Error) {}
+    
+    func response(forProposal proposal: VisitProposal) -> VisitProposalResponse {
+        if proposal.url.path == "/cancel" {
+            return .reject
+        }
```

Comment:
Maybe this is part of another PR you have in mind, but we should discourage inspecting the path to make decisions, since multiple path patterns can be tied to a single set of properties — and can be updated remote. Instead, defer to a custom path config property that can be set for any matching patterns.

Reply:
This is purely for testing that the navigation stack is not modified when the delegate returns `reject`. 

> we should discourage inspecting the path to make decisions, since multiple path patterns can be tied to a single set of properties

This is probably out of scope for this PR. We should discuss it offline.

Reply:
Alternatively, we could bundle a path configuration for the tests and use that. Then we wouldn't even need this `TurboNavigationDelegate` subclass because the routing would come for free.

Reply:
Regardless, a custom path configuration property to determine controllers inside of Turbo Navigator / turbo-ios sounds great to me.

Reply:
> Alternatively, we could bundle a path configuration for the tests and use that. Then we wouldn't even need this TurboNavigationDelegate subclass because the routing would come for free.

I'm definitely in favor of this 👍. This is what we do in turbo-android: 
- https://github.com/hotwired/turbo-android/blob/main/turbo/src/main/assets/json/test-configuration.json
- https://github.com/hotwired/turbo-android/blob/main/turbo/src/test/kotlin/dev/hotwire/turbo/nav/TurboNavRuleTest.kt#L68

Reply:
@olivaresf, I think this is the only outstanding issue.

Do you have appetite for it? If not, I'm happy to take a pass today.

Reply:
I added a test path configuration to use for routing `/cancel`.

How do you both feel about routing a `URL` in this test vs. the `VisitProposal` in all the others? I added a comment but want to make sure it isn't making things more confusing.

Once this is resolved this PR is good to merge in!

Reply:
Leaving it as-is works for me. I don't feel strongly about it either way 👍🏻.

Reply:
Taking another look today, we can do this without a path configuration _or_ routing based on URL. We are already customizing the `VisitProposal` directly under test. Let's continue using that pattern for this one as well!

I think that testing the interaction with path configuration is out of scope for this library. That assigns properties to proposals, we shouldn't have to test or verify that at this level of abstraction.